### ISSUE TITLE
Export/access readable Warp config file

### ISSUE THREAD

**Issue by AlecRust** [warpdotdev/warp#3447]
### Describe the solution you'd like?

I'd like to be able to export or otherwise access my Warp config in a readable format (JSON, TOML, YAML etc.) so that I can version the config along with [my other dotfiles](https://github.com/AlecRust/dotfiles), and script the configuration of Warp on a new machine.

I see you [support](https://docs.warp.dev/features/sessions/launch-configurations) saving some config in `~/.warp/launch_configurations/` which is great, I'd just like to be able to save the rest of Warp config too i.e. all the stuff in the main "Settings" panel.

Something like `~/.warp/warp.toml` or `~/.config/warp/config.toml` would be cool!

### Additional context

Surely this has been requested before, but I can't find a similar issue 🤷 

### How important is this feature to you?

4

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by AlecRust** [comment]
Thanks, useful to know I can read the config in that way.

Just for reference, iTerm2 has a command to set the preferences directory:

```sh
defaults write com.googlecode.iterm2.plist PrefsCustomFolder -string "~/projects/dotfiles/iterm2-settings"
```

A config file somewhere in the home directory that just allowed configuring Appearance, Features, Keybindings etc. would be great.

**Comment by nathanchere** [comment]
Not having easy and convenient access to configuration files for a given piece of software usually tells me one of three things about it.

1. it's not a serious project
2. it was written with a 'Windows-first' or 'mobile-first' thinking
3. it has something to hide

**Comment by strelok** [comment]
super strange that you are working on 'one terminal for mac/windows/linux' and it's not possible to have the same config out of the box...

**Comment by mikefrancis** [comment]
Hacking around with this as I'm getting a new machine soon so don't want to have to write all my config down when I need something to refer back to.

```bash
defaults export dev.warp.Warp-Stable -
```

This will dump Warp config to stdout as an XML plist. I imagine there's XML parsing library in Rust, or could use something intermediary to convert from XML to JSON/YAML/TOML etc.

**Comment by leeyyl** [comment]
I want this feature too because I'm on the new machine and I need to configure warp again. It's very annoying...

**Comment by GraemeF** [comment]
It'd be great to be able to add Warp configuration to the dotfiles repo that I manage with [chezmoi](https://www.chezmoi.io/).

**Comment by tonybenoy** [comment]
> Thanks for this feature request! I know you can run the following command to read the settings for Warp (stored in a dictionary on your Mac) `defaults read dev.warp.Warp-Stable`, so as a possible workaround, there may be a way to import that dictionary to a new Mac.
> 
> To anyone else interested in this feature, please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

Is there a way to do this in Linux?

**Comment by jarodtaylor** [comment]
Would love this as well! Setting up a new machine and my chezmoi dotfiles would have loved a `~/.config/warp` directory/config.

**Comment by nemesis** [comment]
Bump.

Would love a one click (or one command!) transfer/copy/restore to or from another machine. I thought that's why I had to sign up for Warp in the first place? ;)

**Comment by viniciusmi00** [comment]
Still nothing? This feature is sorely missed.
Even if it were a simple option like a button to export settings and another to import, instead of syncing everything to the cloud, it would already help a lot, Warp team.

**Comment by alixmix** [comment]
Bump. I'd love to have this for version control reasons and record keeping.

**Comment by lpurgitoryl** [comment]
Adding my two cents, this would be great as I'm exploring Warp's features and say I would like to roll back to config I already know I like.

**Comment by ehehaye** [comment]
I'd love to have this too. Since warp attracts registration through synchronization, I don't have high expectations for it.

**Comment by futuremotiondev** [comment]
Really want this as well so I can keep all my AI config settings consolidated

**Comment by ehehaye** [comment]
It looks like https://docs.warp.dev/terminal/settings/#opening-your-settings-file could resolve this problem?

**Comment by AlecRust** [comment]
> It looks like [docs.warp.dev/terminal/settings#opening-your-settings-file](https://docs.warp.dev/terminal/settings/#opening-your-settings-file) could resolve this problem?

Yeah I've [moved](https://github.com/AlecRust/dotfiles/commit/1e6fb2cebc80dd211841b5a8772fdfd1b5d1287f) to this, seems ok.

Perhaps I should close this issue now if others agree.

**Comment by Pcmhacker-piro** [comment]
Strong +1 on this idea — version-controlled Warp settings would make machine setup and team onboarding much easier.

I’d love to work on this. My idea is to use a TOML-based config like:

text id="v5rzli" ~/.warp/settings.toml 

since Warp already uses TOML elsewhere.

### Proposed plan
1. Add Export Settings… and Import Settings… in Preferences
2. Export current settings to TOML
3. Import + validate TOML and merge settings safely
4. Optional future enhancement: hot reload on file changes

This could also enable repo-managed/team-shared Warp configs and easier CI/devcontainer setup.

Happy to open a draft PR early for feedback if preferred.

**Comment by acarl005** [comment]
Yes, closing this as we now have a TOML settings file
